llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Passes/TileAndDecomposeAttention.cpp

```diff
@@ -74,21 +74,38 @@ static Value computePartialSoftmax(Value qkTranspose, Value currentMax,
   return genericOp.getResult(0);
 }
 
-static Value updateAndScale(Value oldMax, Value newMax, Value oldSum,
-                            Location loc, OpBuilder &builder,
+static Value computeScaleFactor(Value oldMax, Value newMax, Location loc,
+                                OpBuilder &builder,
+                                SmallVectorImpl<Operation *> &ops) {
+  SmallVector<utils::IteratorType> iteratorTypes(1,
+                                                 utils::IteratorType::parallel);
+  auto identityMap = AffineMap::getMultiDimIdentityMap(1, builder.getContext());
+  SmallVector<AffineMap> indexingMaps(2, identityMap);
+  SmallVector<Type> resultTypes{oldMax.getType()};
+  auto genericOp = builder.create<linalg::GenericOp>(
+      loc, resultTypes, ValueRange{newMax}, ValueRange{oldMax}, indexingMaps,
```

Comment:
nit: Replace `resultTypes` with `oldMax.getType()` and drop the ValueRange wrappers.